src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -587,6 +631,82 @@ private IReadOnlyCollection<ProjectGraphEntryPoint> GetGraphEntryPoints(BuildReq
             }
         }
 
+        public async Task HandleBuildResultAsync(
+            BuildRequestConfiguration requestConfiguration,
+            BuildResult buildResult,
+            BuildEventContext buildEventContext,
+            CancellationToken cancellationToken)
+        {
+            ErrorUtilities.VerifyThrowInternalNull(requestConfiguration.Project, nameof(requestConfiguration.Project));
+
+            if (_projectCachePlugins.IsEmpty)
+            {
+                return;
+            }
+
+            // Filter to plugins which apply to the project, if any
+            List<ProjectCacheDescriptor> projectCacheDescriptors = GetProjectCacheDescriptors(requestConfiguration.Project).ToList();
+            if (projectCacheDescriptors.Count == 0)
+            {
+                return;
+            }
+
+            var globalProperties = new Dictionary<string, string>(requestConfiguration.GlobalProperties.Count, StringComparer.OrdinalIgnoreCase);
+            foreach (ProjectPropertyInstance property in requestConfiguration.GlobalProperties)
+            {
+                globalProperties.Add(property.Name, property.EvaluatedValue);
+            }
+
+            List<string> targets = buildResult.ResultsByTarget.Keys.ToList();
```

Comment:
`FileAccessContext` exposes it as a `IReadOnlyList<string>`. I didn't want to expose the `ICollection<string>` directly since that has `Add` methods on it, although I guess I don't know if those would just throw or not. `IReadOnlyList<string>` is a "better" thing to expose, but the question I suppose is whether that's worth the allocation.